packages/bloc/lib/src/bloc.dart

```diff
@@ -10,23 +9,26 @@ import '../bloc.dart';
 /// and transforms them into a `Stream` of `States` as output.
 /// {@endtemplate}
 abstract class Bloc<Event, State> extends Stream<State> implements Sink<Event> {
-  final PublishSubject<Event> _eventSubject = PublishSubject<Event>();
+  final _states = StreamController<State>();
+  final _events = StreamController<Event>.broadcast();
```

Comment:
Just curious, why does this need to broadcast?

Reply:
I'm confused by this as well...

Reply:
Thanks. Removed unnecessary broadcast stream. It effects `triggers onError from add` from `bloc_test.dart` file. Looks like because broadcast and single subscription streams contains different error message which it test expect. It is funny.

Broadcast stream: `Cannot add new events after calling close`
Single subscription stream: `Cannot add event after closing`